build-system/npm-publish/write-package-files.js

```diff
@@ -251,4 +272,34 @@ async function main() {
   await copyCssToRoot();
 }
 
-main();
+/**
+ * Do some stuff, then return
+ * @param {string} extensionName
+ * @param {string} ampVer
+ * @param {string} extensionVer
+ * @return {Promise<void>}
+ */
+async function writePackageFiles(extensionName, ampVer, extensionVer) {
+  extension = extensionName;
+  packageName = getNameWithoutComponentPrefix(extensionName);
+  ampVersion = ampVer;
+  extensionVersion = extensionVer;
+  dir = getDir(extensionName, extensionVersion);
+  return main();
+}
+
+if (require.main === module) {
+  const [commonExtName, ampVer, extensionVer] = process.argv.slice(2);
+  extension = getNameWithoutComponentPrefix(commonExtName);
+  ampVersion = ampVer;
+  extensionVersion = extensionVer;
+  dir = getDir(extension, extensionVersion);
+  main();
+}
```

Comment:
I don't understand this change, i.e. why `extension, ampVersion, extensionVersion` aren't consts at the top. Can you explain?

Reply:
This is to make it easier to test. If the variables are extracted from `argv` they will be `undefined` when the tests are run. This just initializes the variables from `argv` if the file is called explicitly.

Reply:
could it be easier to pass these as arguments instead of them being set as "global" variables?

Reply:
+1, to pass these as args in `writePackageFiles` and other functions while keeping `main` the same.

Other option is to push args to `process.argv` in the test before importing this module.

Reply:
Yeah, I thought about that but that could have side effects for other tests

Reply:
I've gone ahead and made the change @kvchari suggested but I'll leave it to you to resolve this @estherkim

Reply:
Almost there. Can you test without needing this check `if (require.main === module)` and removing the `_writePackageFiles` wrapper?

Ideally main would look mostly the same.

Reply:
> Yeah, I thought about that but that could have side effects for other tests

If you're worried about that, you can push in a `before` hook and remove in an `after` hook. 

https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#before--after-hooks

Reply:
Yeah, but I would prefer to just not modify the global state